pkg/apk/repo.go

```diff
@@ -984,8 +984,8 @@ func maybedqerror(pkgName string, pkgs []*repositoryPackage, dq map[*RepositoryP
 	}
 
 	if len(errs) != 0 {
-		return fmt.Errorf("solving %q: %w", pkgName, errors.Join(errs...))
+		return &ConstraintError{constraint, errors.Join(errs...)}
 	}
 
-	return fmt.Errorf("could not find package %q in indexes", pkgName)
+	return fmt.Errorf("could not find constraint %q in indexes", constraint)
```

Comment:
Feel free to disregard but what about something like this?

```suggestion
	return fmt.Errorf("could not solve for constraint %q in indexes", constraint)
```